### PR TITLE
drivers: cc13x2 cc26x2: pinctrl: add support for drive strength and edge detection

### PR DESCRIPTION
**Pull request by ghost**
This change set introduces support for additional features of the CC13/26x2 IOC peripheral, namely configuring drive strength and edge detection.

This can then be used in future general purpose timer based drivers, e.g. for (edge) counting or PWM. Such drivers are currently prepared on our side but not yet ready for upstreaming.

@vaishnavachath: This prepares for a generalized timer node with a pwm subnode similar to the generalization in a prior change set for rtc. You may remember that change set. Preview: https://github.com/fgrandel/zephyr/commits/feat/cc13xx_cc26xx_pwm - this will be the next PR as soon as this one has been merged.

### CODE REVIEW

File: dts/bindings/pinctrl/ti,cc13xx-cc26xx-pinctrl.yaml [hunk 1]
```diff
@@ -87,3 +99,16 @@ child-binding:
       type: array
       description: |
         CC13XX/CC26XX pin's configuration (IO pin, IO function).
+
+    drive-strength:
+      enum:
+        - 2
+        - 4
+        - 8
+      default: 2
```
Comment (gmarull) [inline dts/bindings/pinctrl/ti,cc13xx-cc26xx-pinctrl.yaml:120]: default needs to be justified per guidelines

Comment (fgrandel) [reply dts/bindings/pinctrl/ti,cc13xx-cc26xx-pinctrl.yaml:120]: @gmarull Fixed (maybe?) - Can you have another look?

File: dts/bindings/pinctrl/ti,cc13xx-cc26xx-pinctrl.yaml [hunk 2]
```diff
@@ -112,3 +124,13 @@ child-binding:
           2: min 2 mA (SoC default)
           4: min 4 mA
           8: min 8 mA for for double drive strength IOs, min 4 mA for normal IOs
+
+    input-edge-detect:
```
Comment (gmarull) [inline dts/bindings/pinctrl/ti,cc13xx-cc26xx-pinctrl.yaml:128]: `ti,input-edge-detect`

Comment (fgrandel) [reply dts/bindings/pinctrl/ti,cc13xx-cc26xx-pinctrl.yaml]: Fixed.

**Review by gmarull — APPROVED** [review]

**Comment by fgrandel** [comment]
I can reproduce the test failures locally, the same tests fail on main.

**Review by cfriedt — APPROVED** [review]

**Review by vaishnavachath — APPROVED** [review]

**Pull request merged**